src/main/java/org/semux/gui/SwingUtil.java

```
@@ -195,22 +197,42 @@ public static JTextField editableTextField() {
      * Number string comparator based on its value.
      */
     public static final Comparator<String> NUMBER_COMPARATOR = (o1, o2) -> {
-        return Double.compare(Double.parseDouble(o1), Double.parseDouble(o2));
+        try {
+            return Double.compare(NumberFormat.getInstance().parse(o1).doubleValue(),
+                    NumberFormat.getInstance().parse(o2).doubleValue());
+        } catch (ParseException e) {
+            logger.error("NumberFormatException", e);
+        }
+        // we should never come here -> work or exception
+        return 0;
```

`important` this is a library class; if you don't know how to deal with exception, you should throw instead of catch and return. From the caller's perspective, they got 0 return which would mean the two parameters are equal, while there is actually an exception.   YOUR COMMENTS ARE WRONG!

Also, your log error is not descriptive; exception name and message will be printed out by SLF4J. Did you test the exception case?

Similar issues below.

---

Throws a unchecked NumberFormatException 
Contract of functionalInterface does not permit to throw checkedExceptions.